src/Core/Math/LinearAlgebra.hpp

```diff
@@ -156,15 +156,15 @@ namespace Ra
             inline bool checkRange( const Vector_& v, const Scalar& min, const Scalar& max );
 
             /// Get two vectors orthogonal to a given vector.
-            inline void getOrthogonalVectors( const Vector3& fx, Vector3& fy, Vector3& fz );
+            inline void getOrthogonalVectors( const Vector3& fx, Eigen::Ref<Vector3> fy, Eigen::Ref<Vector3> fz );
 
             /// Get the angle between two vectors. Works for types where the cross product is
             /// defined (i.e. 2D and 3D vectors).
             template<typename Vector_>
             inline Scalar angle( const Vector_& v1, const Vector_& v2 );
 
             /// @return the projection of point on the plane define by plane and planeNormal
-            inline Vector3 projectOnPlane(const Vector3& planePos, const Vector3 planeNormal, const Vector3& point);
+            inline Vector3 projectOnPlane(const Vector3& planePos, const Vector3& planeNormal, const Vector3& point);
```

Comment:
Then he says:

> 18:48 <@ChriSopht> use Ref<const ...> when it is semantically necessary 

Again it depends if you plan to take benefit from `Eigen::Ref` or not. Do you believe the non-const function will be called will `mymatrix.col()` while the const function will always be called with an actual `Vector3d` ?

Another option would be to provide both prototypes (i.e. `const Vector3d&` and `const Eigen::Ref<Vector3>`), but I'm not sure the compiler would manage do disambiguate between both cases.